src/argman.rs

```diff
@@ -0,0 +1,709 @@
+        }
+    }
+
+    pub fn get(&self, arg_name: &str) -> &str {
+        self._common_get(arg_name);
+
+        match self.args_help.get(arg_name).unwrap().arg_type {
+            ArgType::ArgStr => {
+                return &self.args.get(arg_name).unwrap()[..];
+            },
+            _ => panic!("get is being used for {}, which is not defined as a str arg", arg_name),
+        }
+    }
+
+    pub fn get_by_category(&self, category: &str, arg_name: &str) -> &str {
+        if !self.args_help.contains_key(arg_name) {
+            panic!("Argument {} is not defined.", arg_name);
+        }
+
+        match self.args_help.get(arg_name).unwrap().arg_type {
+            ArgType::ArgMapStr => {
+
+                if !self.args_multi_map.contains_key(arg_name) {
+                    panic!("Argument {} is not set.", arg_name);
+                }
+
+                if !self.args_multi_map.get(arg_name).unwrap().contains_key(category) {
+                    panic!("no {} category for argument {}", category, arg_name);
+                }
+
+                return &self.args_multi_map.get(arg_name).unwrap().get(category).unwrap()[..];
+            },
+            _ => panic!("get is being used for {}, which is not defined as a map arg", arg_name),
+        }
+    }
+
+    pub fn get_bool(&self, arg_name: &str) -> bool {
+        self._common_get(arg_name);
+
+        match self.args_help.get(arg_name).unwrap().arg_type {
+            ArgType::ArgBool => {
+                let str_val = &self.args.get(arg_name).unwrap()[..];
+                match str_val {
+                    "0" => return false,
+                    "1" => return true,
+                    _ => panic!("Argument {} is a bool and can only be 0 or 1"),
+                }
+            },
+            _ => panic!("get_bool is being used for {}, which is not defined as a bool arg", arg_name),
+        }
+    }
+
+    pub fn get_multi(&self, arg_name: &str) -> &Vec<String> {
+        if !self.args_help.contains_key(arg_name) {
+            panic!("Argument {} is not defined.", arg_name);
+        }
+
+        if self.args_multi.get(arg_name).is_none() {
+            if self.args.get(arg_name).is_none() {
+                panic!("Argument {} is not set.", arg_name);
+            } else {
+                panic!("Argument {} is an argument that cannot be repeated, try 'g_args.get(\"{}\")'.", arg_name, arg_name);
+            }
+        }
+
+        return self.args_multi.get(arg_name).unwrap();
+    }
+
+    pub fn dev_print_selected_args(&self) {
+        println!("\nThe following args were selected:\n");
+        for (name, arg) in &self.args {
+            println!("{}: {:?}", name, arg);
+        }
+        println!("\nThe following args_multi were selected:\n");
+        for (name, arg) in &self.args_multi {
+            println!("{}: {:?}", name, arg);
+        }
+    }
+}
+
+// Tests: Let's make sure ArgMan behaves the way is supposed to in a way that's simple to read
+
+#[test]
```

Comment:
Hmm, normal practice is to put tests inside a #[cfg(test)] mod tests {} module so that they aren't built in release builds. I dunno if this does anything when there are no utility functions, but might as well keep it consistent. Also, if you're gonna add tests, would you mind adding a .travis.yml that runs them? (then I'll turn travis on)

Reply:
1) Done, I forgot about that in rust, thanks.

2) Not sure I understand, sorry. What is "this" in this context? Consistent with what?

3) Sure, just new to editing travis files, so don't hesitate to correct me.